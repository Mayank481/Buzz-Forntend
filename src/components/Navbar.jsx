import { Link } from 'react-router-dom';
import { handleLogout } from '../services/authServices';

export default function Navbar({ user }) {
  return (
    <>
      <nav className="navbar  navbar-light bg-light p-0">
        <div className="container-fluid">
          <div>
            <Link to={'/'}>
         
              <div id='logo-img'></div>
            </Link>
          </div>

          <div className="d-flex position-relative">
            <Link to="/profile" className="nav-link text-dark">
              <div className="d-flex">
                <div className="">
                  {'picture_url' in user ? (
                    <img
                      src={user.picture_url}
                      className="card-img-top small-round-pic  round-img"
                      alt="..."
                    />
                  ) : (
                    <i className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"></i>
                  )}
                </div>
                <div className=" align-items-center ms-2 resp-hide" data-testid="userInfo">
                  {'firstname' in user
                    ? user.firstname + ' ' + user.lastname
                    : 'Edit Profile'}
                </div>
              </div>
            </Link>

            <div className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2">
              <Link to={'/'} className="text-dark">
                <i className="fa-solid fa-house"></i>
              </Link>
            </div>
            <div className="position-relative d-flex ">
              <Link
                to={'/friends'}
                className="d-flex align-items-center  ms-2 round-img border rounded-circle icon-bg text-dark p-2 "
              >
                <i className="fa-solid fa-user"></i>
              </Link>
              <div className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0">
                {user.friends.myFriendRequests.length}
              </div>
            </div>
            <div
              onClick={handleLogout}
              className="d-flex align-items-center ms-2 me-3 round-img border rounded-circle icon-bg text-dark p-2 pointer"
            >
              <i className="fa fa-sign-out "></i>
            </div>
          </div>
        </div>
      </nav>
    </>
  );
}
