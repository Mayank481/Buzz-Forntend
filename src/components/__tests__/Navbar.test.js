import { render, screen } from '@testing-library/react';
import { useState } from 'react';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../Navbar';
import renderer from 'react-test-renderer';


test('should render Navbar Panel', () => {
    const user = { firstname: "John", lastname: "Doe", friends: { myFriends: [], mySentRequests: [{}], myFriendRequests: [{}] } };
    render(<BrowserRouter> <Navbar user={user} /> </BrowserRouter>)
    const text = screen.queryByTestId('userInfo').innerHTML;
    expect(text).toBe("John Doe");

    const component = renderer.create(<BrowserRouter> <Navbar user={user} /> </BrowserRouter>)
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot();
})