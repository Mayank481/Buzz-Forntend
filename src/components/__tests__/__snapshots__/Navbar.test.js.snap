// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Navbar Panel 1`] = `
Array [
  " ",
  <nav
    className="navbar  navbar-light bg-light p-0"
  >
    <div
      className="container-fluid"
    >
      <div>
        <a
          href="/"
          onClick={[Function]}
        >
          <div
            id="logo-img"
          />
        </a>
      </div>
      <div
        className="d-flex position-relative"
      >
        <a
          className="nav-link text-dark"
          href="/profile"
          onClick={[Function]}
        >
          <div
            className="d-flex"
          >
            <div
              className=""
            >
              <i
                className="fa-solid fa-user fa-2x card-img-top small-round-pic  round-img bg-warning d-flex justify-content-center align-items-center"
              />
            </div>
            <div
              className=" align-items-center ms-2 resp-hide"
              data-testid="userInfo"
            >
              John Doe
            </div>
          </div>
        </a>
        <div
          className="d-flex align-items-center ms-2 round-img border rounded-circle icon-bg text-dark p-2"
        >
          <a
            className="text-dark"
            href="/"
            onClick={[Function]}
          >
            <i
              className="fa-solid fa-house"
            />
          </a>
        </div>
        <div
          className="position-relative d-flex "
        >
          <a
            className="d-flex align-items-center  ms-2 round-img border rounded-circle icon-bg text-dark p-2 "
            href="/friends"
            onClick={[Function]}
          >
            <i
              className="fa-solid fa-user"
            />
          </a>
          <div
            className="round-img bg-danger p-1 text-white incoming position-absolute bottom-50 end-0"
          >
            1
          </div>
        </div>
        <div
          className="d-flex align-items-center ms-2 me-3 round-img border rounded-circle icon-bg text-dark p-2 pointer"
          onClick={[Function]}
        >
          <i
            className="fa fa-sign-out "
          />
        </div>
      </div>
    </div>
  </nav>,
  " ",
]
`;
